lib/galaxy/webapps/galaxy/buildapp.py

```diff
@@ -292,6 +293,11 @@ def populate_api_routes( webapp, app ):
                             path_prefix='/api',
                             collection={ 'sniffers': 'GET', 'mapping': 'GET', 'converters': 'GET', 'edam_formats': 'GET' },
                             parent_resources=dict( member_name='datatype', collection_name='datatypes' ) )
+    webapp.mapper.resource( 'datatype',
+                            'datatypes',
+                            path_prefix='/api',
+                            collection={ 'sniffers': 'GET', 'mapping': 'GET', 'converters': 'GET', 'edam_data': 'GET' },
+                            parent_resources=dict( member_name='datatype', collection_name='datatypes' ) )
```

Comment:
I think that instead of invoking again `webapp.mapper.resource()` here, you should just add `'edam_data': 'GET'` at line 294.